python/ray/train/tests/test_state.py

```diff
@@ -155,6 +159,63 @@ def test_state_actor_api():
         assert run_info == info_list[i]
 
 
+class TestConfig(BackendConfig):
+    @property
+    def backend_cls(self):
+        return TestBackend
+
+
+class TestBackend(Backend):
+    def on_start(self, worker_group: WorkerGroup, backend_config: TestConfig):
+        pass
+
+    def on_shutdown(self, worker_group: WorkerGroup, backend_config: TestConfig):
+        pass
+
+
+def test_state_manager(ray_start_gpu_cluster):
+    os.environ["RAY_TRAIN_ENABLE_STATE_TRACKING"] = "0"
+    e = BackendExecutor(
+        backend_config=TestConfig(), num_workers=4, resources_per_worker={"GPU": 1}
+    )
+    e.start()
```

Comment:
Nit: we can just create a `WorkerGroup` directly instead of using `BackendExecutor`.

Reply:
Changed to `WorkerGroup`. 

I was originally trying to also test the session initialization in this test. But since we already covered that in the end-2-end test, let's go with `WorkerGroup` instead.